thrall/app/lib/ElasticSearch.scala

```diff
@@ -62,40 +63,56 @@ object ElasticSearch extends ElasticSearchClient with ImageFields {
-      andFilter(
-        missingOrEmptyFilter("exports"),
-        missingOrEmptyFilter(identifierField(Config.persistenceIdentifier)),
-        boolFilter.mustNot(termFilter(editsField("archived"), true)),
-        boolFilter.mustNot(termFilter(usageRightsField("category"), "staff-photographer")),
-        boolFilter.mustNot(termFilter(usageRightsField("category"), "contract-photographer")),
-        boolFilter.mustNot(termFilter(usageRightsField("category"), "commissioned-photographer"))
-      )
+  def deleteFilterQuery(id: String) = filteredQuery(
+    boolQuery.must(matchQuery("_id", id)),
+    andFilter(
+      missingOrEmptyFilter("exports"),
+      missingOrEmptyFilter(identifierField(Config.persistenceIdentifier)),
+      boolFilter.mustNot(termFilter(editsField("archived"), true)),
+      boolFilter.mustNot(termFilter(usageRightsField("category"), "staff-photographer")),
+      boolFilter.mustNot(termFilter(usageRightsField("category"), "contract-photographer")),
+      boolFilter.mustNot(termFilter(usageRightsField("category"), "commissioned-photographer"))
     )
+  )
 
-    val deleteQuery = client
+  private def deleteImageQuery(id: String) =
+    // we could use a `prepareDelete(imagesAlias, imageType, id), but we should
+    // never really expose the ability to delete without applying our rules first
+    client
       .prepareDeleteByQuery(imagesAlias)
       .setTypes(imageType)
-      .setQuery(q)
+      .setQuery(deleteFilterQuery(id))
+      .executeAndLog(s"Attempting to delete image $id")
+
+  def deleteImage(id: String)
+                       (implicit ex: ExecutionContext):
+                        Future[Either[ImageDeletedResponse, ImageNotDeletableResponse]] = {
 
-    // search for the image first, and then only delete and succeed
-    // this is because the delete query does not respond with anything useful
+    // We search for the image first as the `prepareDelete` doesn't return whether
+    // it has delete anything or not. We need this info further up the chain to know
+    // whether to delete the physical images or not.
     // TODO: is there a more efficient way to do this?
     client
       .prepareCount()
-      .setQuery(q)
+      .setQuery(deleteFilterQuery(id))
       .executeAndLog(s"Searching for image to delete: $id")
       .flatMap { countQuery =>
         val deleteFuture = countQuery.getCount match {
-          case 1 => deleteQuery.executeAndLog(s"Deleting image $id")
-          case _ => Future.failed(ImageNotDeletable)
+          case 1 => deleteImageQuery(id)
+          case _ => Future.failed(ImageNotDeletableResponse(id))
         }
         deleteFuture
           .incrementOnSuccess(deletedImages)
-          .incrementOnFailure(failedDeletedImages) { case ImageNotDeletable => true }
+          .incrementOnFailure(failedDeletedImages) { case _:ImageNotDeletableResponse => true }
+
+        deleteFuture map { deleteResponse =>
+          Left(ImageDeletedResponse(id))
```

Comment:
Normally, `Left` is used for failures and `Right` for success. Is the inversion here intentional?